File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) == 1:
            return False
        stack = []
        for c in s:
            if c == '(' or c == '{' or c == '[':
                stack.append(c)
            else:
                open_bracket = stack.pop()
                if (c == ')' and open_bracket != '(') or (c == '}' and open_bracket != '{') or (c == ']' and open_bracket != '['):
                    return False
        return True

File: Data Structures & Algorithms/validate-parentheses/submission-1.java
class Solution {
    public boolean isValid(String s) {
        if (s.length() == 1) {
            return false;
        }
        Stack<Character> openBrackets = new Stack<Character>();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(' || s.charAt(i) == '[' || s.charAt(i) == '{') {
                openBrackets.push(s.charAt(i));
            } else {
                if (openBrackets.size() == 0) {
                    return false;
                }
                char openBracket = openBrackets.pop();
                if (openBracket == '(' && s.charAt(i) != ')' || openBracket == '[' && s.charAt(i) != ']' || openBracket == '{' && s.charAt(i) != '}') {
                    return false;
                }
            }
        }
        return openBrackets.size() == 0;
    }
}
